Warnings in HAN collectives

OMPI main, built on Mac:
```shell
coll_han_gather.c: In function 'mca_coll_han_gather_intra_simple':
coll_han_gather.c:409:25: warning: pointer of type 'void *' used in arithmetic [-Wpointer-arith]
  409 |             sbuf = rbuf + rextent * (ptrdiff_t)rcount * w_rank;
      |                         ^
coll_han_component.c: In function 'han_register':
coll_han_component.c:440:18: warning: declaration of 'topo_lvl' shadows a previous local [-Wshadow]
  440 |         for (int topo_lvl = 0 ; topo_lvl < GLOBAL_COMMUNICATOR ; topo_lvl++) {
      |                  ^~~~~~~~
coll_han_component.c:251:16: note: shadowed declaration is here
  251 |     TOPO_LVL_T topo_lvl;
      |                ^~~~~~~~
```

--- a/ompi/communicator/comm.c
+++ b/ompi/communicator/comm.c
@@ -3,7 +3,7 @@
  * Copyright (c) 2004-2005 The Trustees of Indiana University and Indiana
  *                         University Research and Technology
  *                         Corporation.  All rights reserved.
- * Copyright (c) 2004-2019 The University of Tennessee and The University
+ * Copyright (c) 2004-2023 The University of Tennessee and The University
  *                         of Tennessee Research Foundation.  All rights
  *                         reserved.
  * Copyright (c) 2004-2022 High Performance Computing Center Stuttgart,
@@ -2215,10 +2215,10 @@ int ompi_comm_get_rprocs (ompi_communicator_t *local_comm, ompi_communicator_t *
     int rc = OMPI_SUCCESS;
     int local_rank, local_size;
     ompi_proc_t **rprocs=NULL;
-    int32_t size_len;
+    size_t size_len;
     int int_len=0, rlen;
     pmix_data_buffer_t *sbuf=NULL, *rbuf=NULL;
-    void *sendbuf=NULL;
+    char *sendbuf=NULL;
     char *recvbuf;
     ompi_proc_t **proc_list = NULL;
     int i;

--- a/ompi/mca/coll/base/coll_base_alltoall.c
+++ b/ompi/mca/coll/base/coll_base_alltoall.c
@@ -3,7 +3,7 @@
  * Copyright (c) 2004-2005 The Trustees of Indiana University and Indiana
  *                         University Research and Technology
  *                         Corporation.  All rights reserved.
- * Copyright (c) 2004-2021 The University of Tennessee and The University
+ * Copyright (c) 2004-2023 The University of Tennessee and The University
  *                         of Tennessee Research Foundation.  All rights
  *                         reserved.
  * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
@@ -274,7 +274,7 @@ int ompi_coll_base_alltoall_intra_bruck(const void *sbuf, int scount,
     tmpbuf = tmpbuf_free - gap;
 
     /* Step 1 - local rotation - shift up by rank */
-    err = ompi_datatype_sndrcv (sbuf + ((ptrdiff_t) rank * scount * sext),
+    err = ompi_datatype_sndrcv ((char*)sbuf + ((ptrdiff_t) rank * scount * sext),
                                 (int32_t) (size - rank) * scount,
                                 sdtype,
                                 tmpbuf,
@@ -285,7 +285,7 @@ int ompi_coll_base_alltoall_intra_bruck(const void *sbuf, int scount,
     }
 
     if (rank != 0) {
-        err = ompi_datatype_sndrcv (sbuf,
+        err = ompi_datatype_sndrcv ((char*)sbuf,
                                     (int32_t) rank * scount,
                                     sdtype,
                                     tmpbuf + ((ptrdiff_t) (size - rank) * rcount * rext),

--- a/ompi/mca/coll/cuda/coll_cuda_reduce.c
+++ b/ompi/mca/coll/cuda/coll_cuda_reduce.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2004-2015 The University of Tennessee and The University
+ * Copyright (c) 2004-2023 The University of Tennessee and The University
  *                         of Tennessee Research Foundation.  All rights
  *                         reserved.
  * Copyright (c) 2014-2015 NVIDIA Corporation.  All rights reserved.
@@ -37,7 +37,6 @@ mca_coll_cuda_reduce(const void *sbuf, void *rbuf, int count,
     int rank = ompi_comm_rank(comm);
     ptrdiff_t gap;
     char *rbuf1 = NULL, *sbuf1 = NULL, *rbuf2 = NULL;
-    const char *sbuf2;
     size_t bufsize;
     int rc;
 
@@ -53,7 +52,6 @@ mca_coll_cuda_reduce(const void *sbuf, void *rbuf, int count,
             return OMPI_ERR_OUT_OF_RESOURCE;
         }
         mca_coll_cuda_memcpy(sbuf1, sbuf, bufsize);
-        sbuf2 = sbuf; /* save away original buffer */
         sbuf = sbuf1 - gap;
     }
 

--- a/ompi/mca/coll/han/coll_han_component.c
+++ b/ompi/mca/coll/han/coll_han_component.c
@@ -437,7 +437,7 @@ static int han_register(void)
         /*
          * Default values
          */
-        for (int topo_lvl = 0 ; topo_lvl < GLOBAL_COMMUNICATOR ; topo_lvl++) {
+        for (topo_lvl = 0 ; topo_lvl < GLOBAL_COMMUNICATOR ; topo_lvl++) {
             cs->mca_sub_components[coll][topo_lvl] = TUNED;
         }
         cs->mca_sub_components[coll][GLOBAL_COMMUNICATOR] = HAN;

